lib/webapp.js

```diff
@@ -134,6 +149,51 @@ DataserviceContext.prototype = {
   makeErrorObject: zluxUtil.makeErrorObject
 };
 
+function createDataserviceStorage() {
+  return process.clusterManager.getStorage().then( function(clusterStorage) {
+    
+    clusterStorage.getStorage = function () {
+      return process.clusterManager.getStorage().then((storage) => {
+        clusterStorage.storageDict = storage.storageDict;
+        return storage.storageDict;
+      });
+    }
+
+    clusterStorage.setStorageValue = function (id, value) {
+      clusterStorage.storageDict[id] = value;
+      process.clusterManager.setStorageValue(id, value);
+    }
+
+    clusterStorage.setStorage = function (dict) {
+      clusterStorage.storageDict = dict;
+      process.clusterManager.setStorage(dict);
+    }
+
+    clusterStorage.addStorage = function (dict) {
+      clusterStorage.storageDict = { ...clusterStorage.storageDict, ...dict };
+      process.clusterManager.addStorage(dict);
+    }
+
+    clusterStorage.getStorageValue = function (id) {
+      process.clusterManager.getStorageValue(id).then((value) => {
+        return value;
+      });
+    }
+
+    clusterStorage.deleteStorageValue = function (id) {
+      delete clusterStorage.storageDict[id];
```

Comment:
This seems wrong to me, because it assumes there are 2 storage objects at this time. There should be 1: on the cluster manager

Reply:
Yep, there are two. My misunderstanding! See my reply.